editor/test/effects.js

```diff
@@ -321,6 +322,96 @@ describe( 'effects', () => {
 		} );
 	} );
 
+	describe( '.TRASH_POST', () => {
+		it( 'should dispatch an removal notice for trashing a post as well as a trash post success action.', () => {
+			const apiCall = ( post ) => ( {
+				getPostTypeModel: function( postType ) { // eslint-disable-line no-unused-vars
+					return function( postObject ) { // eslint-disable-line no-unused-vars
+						return {
+							destroy: function() {
+								return this;
+							},
+							then: function( callback, errorHandler ) {
+								if ( post !== 'error' ) {
+									callback();
+								} else {
+									errorHandler( 'error' );
+								}
+							},
+						};
+					};
+				},
+			} );
+			const post = {
+				id: 1,
+				title: {
+					raw: 'A History of Pork',
+				},
+				content: {
+					raw: '',
+				},
+				status: 'draft',
+			};
+			wp.api = apiCall( post );
```

Comment:
What do you think about stubing `wp.api.getPostTypeModel` method instead? We could verify if `getCurrentPostType` selector returns proper object from the state. This would also allow us to make  `getPostTypeModel` mock much shorter.